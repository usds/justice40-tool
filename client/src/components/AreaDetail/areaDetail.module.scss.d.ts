declare namespace MapModuleScssNamespace {
  export interface IMapModuleScss {
    areaDetailContainer: string;
    categorization:string;
    communityOfFocus:string;
    censusRow:string;
    censusLabel:string;
    censusText: string;
    feedbackLink:string;
    isInFocus:string;
    versionInfo: string;
    showThresholdExceed:string;
    showCategoriesExceed: string;
    categoryHeader:string;
    sendFeedbackLink: string;
    sendFeedbackBtn: string;
    buttonContainer: string;
    buttonText: string;
    buttonImage: string;
    categorySpacer: string;
    testSignals: string;
    prioCopy: string;
    federalRecogTribes: string;
  }
}

declare const MapModuleScssModule: MapModuleScssNamespace.IMapModuleScss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: MapModuleScssNamespace.IMapModuleScss;
};

export = MapModuleScssModule;
