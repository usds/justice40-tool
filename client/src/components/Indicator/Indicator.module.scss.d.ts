declare namespace IndicatorNamespace {
    export interface IIndicatorScss {
      indicatorBoxMain:string;
      indicatorBoxAdditional:string;
      indicatorRow:string;
      indicatorName:string;
      indicatorValueCol:string;
      indicatorValueRow:string;
      indicatorValue:string;
      disIndicatorValue:string;
      indicatorSuperscript:string;
      indicatorInfo:string;
      info:string;
      infoTilde: string;
      indicatorValueSubText:string;
      indicatorDesc:string;
      disadvantagedIndicator:string;
    }
  }

declare const IndicatorScssModule: IndicatorNamespace.IIndicatorScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: IndicatorNamespace.IIndicatorScss;
  };

  export = IndicatorScssModule;
