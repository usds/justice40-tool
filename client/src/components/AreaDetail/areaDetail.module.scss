@use '../../styles/design-system.scss' as *;
@import "../utils.scss";

$sidePanelLabelFontColor: #171716;

@mixin sidePanelLabelStyle {
  font-size: medium;
  color: $sidePanelLabelFontColor;
  font-weight: 600;
}

.versionInfo {
  padding: 2rem 1rem 2rem 1.2rem;
  text-align: center;
  font-size: medium;
  color: #A0A3A3;
}

.areaDetailContainer {
  display: flex;
  flex-direction: column;
}

.categorization {
  display: flex;
  flex-direction: column;
  align-items: center;
  // @include u-padding-bottom(3);

  .isInFocus {
    padding: .5rem 1rem .25rem 1.2rem;
    font-size: small;
    font-weight: bold;
    @include u-margin-top(2);
  }

  .communityOfFocus {
    display: flex;

    h3 {
      margin-top: 0;
      margin-bottom: 0;
    }

  }

  .prioCopy{
    @include u-padding-left(2);
    @include u-padding-right(2);
    @include u-padding-top(2);

  }

  .showCategoriesExceed {
    font-size: small;
    @include u-margin-top('05');
  }

}

.sendFeedbackLink {
  @include u-margin-top(2);
  @include u-margin-left(2);
  @include u-margin-bottom(2);

  .sendFeedbackBtn{
    @include u-text("blue-70v");
    @include u-bg("yellow-20v");
    height: 40px; 
    
    &:hover {
      @include u-bg("yellow-20");
      @include u-text("gray-90");
    }

    .buttonContainer{
      display: flex;

      .buttonText{
        @include u-margin-right(1);
      }

      .buttonImage{
        width: 21px;
        margin-top: -3px;

        filter: invert(13%) sepia(76%) saturate(5142%) hue-rotate(192deg) brightness(80%) contrast(106%);
      }
    }
  }
}

//Census row styles
.censusRow {
  display: flex;
  flex-direction: column;
  list-style: none;
  margin: 0;
  padding: 1.2rem 1rem 0 1.2rem;

  .censusLabel {
    @include sidePanelLabelStyle;
  }

  .censusText {
    font-size: medium;
  }  
}

.categoryHeader {
  display: flex;
  justify-content: space-between;
  text-transform: uppercase;
  font-size: small;
  @include u-bg('gray-cool-5');
  @include u-padding-left(2.5);
  @include u-padding-right(2);
}

.categorySpacer {
  @include typeset('sans', '2xs', 2);
  @include u-text('bold'); 

  margin-left: -20px;
  @include u-padding-top(1);
  @include u-padding-bottom(1);
  @include u-padding-left(2.5);
}

.testSignals{
  padding-left: 1rem;
  padding-bottom: 1rem;
}
