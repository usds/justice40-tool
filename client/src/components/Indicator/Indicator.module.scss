@use '../../styles/design-system.scss' as *;
@import "../utils.scss";

@mixin indicatorPadding {
  @include u-padding-left("05");
  @include u-padding-right("05");
}

@mixin indicator {
  display: flex;
  flex-direction: column;
  @include u-padding-top(1.5);
  @include u-padding-bottom(1.5);
  
  .indicatorRow {
    display: flex;
    justify-content: space-between;
    
    @media screen and (max-width: $mobileBreakpoint) {
      flex: 1 0 40%;
      align-self: inherit;
      padding-left: 3rem;
      padding-top: 1rem;
    }
    
    .indicatorName {
      flex-basis: 55%;
      display: flex;
      flex-direction: column;
      @include typeset('sans', '2xs', 2);
      @include u-text('bold'); 
      
      .indicatorDesc {
        @include typeset('sans', '3xs', 2); 
        @include u-text('thin'); 
        max-width: 12rem;    
        @include u-margin-top(0);
        @media screen and (max-width: 1024px) {
          max-width: 80%;
        }
      }
    }
    
    .indicatorValueCol {
      display: flex;
      flex-direction: column;
      @include typeset('sans', '2xs', 2);
      @include u-text('bold'); 
      width: 40%;

      .indicatorValueRow {
        display: flex;
        align-self: end;

        .indicatorValue {
          @include indicatorPadding();
        }
        
        .disIndicatorValue {
          @include indicatorPadding();
          color: white;
          background-color: $disadvantagedDotColor;
        }

        .indicatorInfo {
          margin-bottom: -.375rem;
          margin-top: -2px;
          @include u-margin-right('05');

          img.info {
            max-width: none;
            height: 1.2rem;
            width: 1.2rem;
          }

          .infoTilde {
            vertical-align: super;
          }
        }
      }
      
      .indicatorValueSubText{
        @include indicatorValueSubTextContainer();
      }
    }

  }
}

//Indicator box styles
.indicatorBoxMain {
  @include indicator;
  @include u-color('gray-warm-90');
}
