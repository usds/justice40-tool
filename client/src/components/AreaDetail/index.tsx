import AreaDetail from './AreaDetail';

export default AreaDetail;

