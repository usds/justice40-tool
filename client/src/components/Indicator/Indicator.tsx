import React from 'react';
import {useIntl} from 'gatsby-plugin-intl';
import ReactTooltip from 'react-tooltip';

import {indicatorInfo, indicatorType} from '../AreaDetail/AreaDetail';

import * as styles from './Indicator.module.scss';
import * as constants from '../../data/constants';
import * as EXPLORE_COPY from '../../data/copy/explore';

// @ts-ignore
import infoIcon from '/node_modules/uswds/dist/img/usa-icons/info.svg';

interface IIndicator {
  indicator: indicatorInfo,
  isImpute?: boolean,
  population?: number | string,
}
interface IIndicatorValueSubText {
  type: indicatorType,
  value: number | null | boolean,
  isAboveThresh: boolean,
  threshold: number,
}

interface IIndicatorValue {
  type: indicatorType,
  displayStat: number | null,
}

/**
 * This component will render an info icon in the indicator value
 *
 * @return {JSX.Element}
 */
export const IndicatorInfoIcon = ({isImpute, population}: Omit<IIndicator, 'indicator'>) => {
  const intl = useIntl();
  let showTilde = false;

  const getToolTipCopy = () => {
    if (population === constants.MISSING_DATA_STRING) {
      return intl.formatMessage(EXPLORE_COPY.LOW_INCOME_TOOLTIP.IMP_YES_POP_NULL);
    } else if (population !== constants.MISSING_DATA_STRING && isImpute) {
      showTilde = true;
      return intl.formatMessage(EXPLORE_COPY.LOW_INCOME_TOOLTIP.IMP_YES_POP_NOT_NULL);
    } else {
      return null;
    }
  };

  /**
   * This library react-tooltip creates random DOM ID which will not allow for snapshot testing as
   * the IDs change on each build. Due to time constraints, we simply removed the AreaDetails test.
   * The AreaDetails component is made up of sub component and each sub component has tests so this
   * is low risk.
   *
   * This is a temporary solution. Some longer terms solutions may be
   * 1. Remove this library and get the USWDS tool tip to work
   * 2. Re-factor the areaDetail.tests.tsx snapshot tests to do more DOM assertions rather than snapshots
   * 3. Some combination of the two.
   */
  return (
    <>
      <ReactTooltip
        id="lowIncomeIcon"
        multiline={true}
      />
      <img
        data-for="lowIncomeIcon"
        data-tip={getToolTipCopy()}
        data-iscapture="true"
        className={styles.info}
        src={infoIcon}
        alt={intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_VALUES.IMG_ALT_TEXT.INFO)}
      />
      {showTilde && <span className={styles.infoTilde}>{ ` ~ ` }</span>}
    </>
  );
};

/**
 * This component will determine the sub-text of the indicator's value, some examples could be
 *   "above 90th percentile"
 *   "below 20 percent"
 *   "data is not available"
 *
 * @param {IIndicatorValueSubText} {}
 * @return {JSX.Element}
 */
export const IndicatorValueSubText = ({type, value, isAboveThresh, threshold}:IIndicatorValueSubText) => {
  if (value === null) {
    return (
      <div>
        {EXPLORE_COPY.SIDE_PANEL_VALUES.UNAVAILBLE_MSG}
      </div>
    );
  } else if (type === 'percent' || type === 'percentile') {
    return (
      <div>
        {
          isAboveThresh ?
          EXPLORE_COPY.SIDE_PANEL_VALUES.ABOVE :
          EXPLORE_COPY.SIDE_PANEL_VALUES.BELOW
        }
        {
          threshold ?
          <IndicatorValue type={type} displayStat={threshold}/> :
          <IndicatorValue type={type} displayStat={90}/>
        }
        {` `}
        {
          type === 'percent' ?
          EXPLORE_COPY.SIDE_PANEL_VALUES.PERCENT :
          EXPLORE_COPY.SIDE_PANEL_VALUES.PERCENTILE
        }
      </div>
    );
  } else {
    return (<></>);
  }
};

/**
 * The react-i18n library allow to add ordinal suffix (st, nd, rd, th) to indicator values.
 * The number and the suffix is a single entity. We are looking to add styling to
 * just the suffix portion if the value is a percentile. This function will add
 * a superscript styling to just the suffix portion of percentile values.
 *
 * The i18n variable named i18nOrdinalSuffix, in the IndicatorValue function defines the
 * various prefixes. The Spanish version of the i18n variable works in a similar manner,
 * however has a differnce. The superscripting is different for Spanish.
 * In Spanish, the suffix is a ".a" and ".o", where only the "a" and "o" are superscripted.
 * This function handles this case.
 *
 * Verbatim from translation team:
 * We suggest changing this to the Spanish ordinal number abbreviation, which is .o for masculine
 * and .a for feminine gendered words. ***Since this ranking applies to the communities of focus,
 * which use a feminine gender in Spanish, we recommend that the th ordinal abbreviation in English
 * be substituted with the feminine ordinal abbreviation in Spanish: .a  throughout the text.
 * E.g., 19th would be 19.a in Spanish and 65th would be 65.a
 *
 * @param {string} indicatorValueWithSuffix
 * @return {string}
 */
export const superscriptOrdinal = (indicatorValueWithSuffix:string) => {
  // Spanish case:
  if (indicatorValueWithSuffix.indexOf('.') !== -1) {
    const ordinalSuffix = indicatorValueWithSuffix.charAt(indicatorValueWithSuffix.length - 1);
    const indicatorValue = indicatorValueWithSuffix.slice(0, -1);

    return <>{indicatorValue}<sup style={{top: '-0.2em'}}>{ordinalSuffix}</sup></>;
  }

  // English case:
  const valueRegEx = /[0-9]{1,2}/;
  const suffixRegEx = /[a-z]{2}/; // ie, (st, nd, rd, th)
  const indicatorValue = valueRegEx.exec(indicatorValueWithSuffix);
  const ordinalSuffix = suffixRegEx.exec(indicatorValueWithSuffix);

  return <>{indicatorValue}<sup style={{top: '-0.2em'}}>{ordinalSuffix}</sup></>;
};

/**
 * This component will return the indicators's value. The value depends on the
 * indicator type. Each type renders a different UI.
 *
 * @return {JSX.Element | null}
 */
export const IndicatorValue = ({type, displayStat}:IIndicatorValue) => {
  const intl = useIntl();

  if (displayStat === null) return <>{constants.MISSING_DATA_STRING}</>;

  if (type === 'percent' || type === 'percentile') {
    // In this case we will show no value and an icon only

    if (type === 'percent') {
      // If the type is percent, return the intl percent format
      return (
        <span>
          {intl.formatNumber(
              displayStat,
              {
                style: 'unit',
                unit: 'percent',
                unitDisplay: 'short',
              },
          )}
        </span>
      );
    } else {
    // If the type is percentile, create the intl ordinal and return it as a superscript
      const i18nOrdinalSuffix: string = intl.formatMessage(
          {
            id: 'explore.map.page.side.panel.indicator.percentile.value.ordinal.suffix',
            // eslint-disable-next-line max-len
            description: `Navigate to the explore the tool page. Click on the map. The side panel will show categories. Open a category. This will define the indicator value's ordinal suffix. For example the st in 91st, the rd in 23rd, and the th in 26th, etc.`,
            defaultMessage: `
        {indicatorValue, selectordinal, 
          one {#st} 
          two {#nd}
          =3 {#rd} 
          other {#th}
        }
        `,
          },
          {
            indicatorValue: displayStat,
          },
      );
      return superscriptOrdinal(i18nOrdinalSuffix);
    }
  } else {
    // when the type === boolean the display stat will be either 100 (true) or 0 (false)
    return displayStat === 0 ?
      EXPLORE_COPY.SIDE_PANEL_SPACERS.NO :
      EXPLORE_COPY.SIDE_PANEL_SPACERS.YES;
  }
};

/**
 * This component will return the list element which will be the indicator row in the side panel
 *
 * @param {IIndicator} indicator
 * @return {JSX.Element}
 */
const Indicator = ({indicator, isImpute, population}:IIndicator) => {
  /**
   * The indicator value could be a number | boolean | null. In all cases we coerce to number
   * before flooring.
   *
   * In the case where indicator.value is a boolean, the displayStat will be either 100 or 0, depending
   * on if indicator.value is true or false respectively.
   *
   * Todo: The way the displayStat handles the boolean indicators should be refactored
   */
  const displayStat = indicator.value !== null ? Math.floor(Number(indicator.value) * 100) : null;

  // If the threshold exists, set it, otherwise set it to the default value
  const threshold = indicator.threshold ? indicator.threshold : constants.DEFAULT_THRESHOLD_PERCENTILE;

  // A boolean to represent if the indicator is above or below the threshold
  const isAboveThresh = displayStat !== null && displayStat >= threshold ? true : false;

  // Show an info icon on the low icome indicator if:
  const showLowIncomeInfoIcon = (
    (indicator.label === 'Low income' && (isImpute)) ||
    (indicator.label === 'Low income' && population === constants.MISSING_DATA_STRING && !isImpute)
  );

  return (
    <li
      className={styles.indicatorBoxMain}
      data-cy={'indicatorBox'}
      data-testid='indicator-box'>
      <div className={styles.indicatorRow}>

        {/* Indicator name and description*/}
        <div className={styles.indicatorName}>
          {indicator.label}
          <div className={styles.indicatorDesc}>
            {indicator.description}
          </div>
        </div>

        {/* Indicator value, icon and subtext */}
        <div className={styles.indicatorValueCol}>
          <div className={styles.indicatorValueRow}>

            {/* Indicator info icon */}
            { showLowIncomeInfoIcon &&
              <div className={styles.indicatorInfo}>
                <IndicatorInfoIcon
                  isImpute={isImpute}
                  population={population}
                />
              </div>
            }

            {/* Indicator value */}
            <div className={indicator.isDisadvagtaged ?
              styles.disIndicatorValue : styles.indicatorValue}
            >
              <IndicatorValue
                type={indicator.type}
                displayStat={displayStat}
              />
            </div>

          </div>

          {/* Indicator sub-text */}
          <div className={styles.indicatorValueSubText}>
            <IndicatorValueSubText
              value={displayStat}
              isAboveThresh={isAboveThresh}
              threshold={threshold}
              type={indicator.type}
            />
          </div>

        </div>
      </div>
    </li>
  );
};

export default Indicator;
