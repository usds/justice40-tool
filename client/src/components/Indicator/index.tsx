import Indicator from './Indicator';

export default Indicator;
