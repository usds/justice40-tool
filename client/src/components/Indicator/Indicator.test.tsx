import * as React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import Indicator, {IndicatorValueSubText, IndicatorValue} from './Indicator';
import {indicatorInfo} from '../AreaDetail/AreaDetail';

describe('rendering of the Indicator', () => {
  it('checks if component renders', () => {
    const highSchool:indicatorInfo = {
      label: 'some label',
      description: 'some description',
      type: 'percent',
      value: .97,
      isDisadvagtaged: true,
      threshold: 20,
    };
    const {asFragment} = render(
        <LocalizedComponent>
          <Indicator indicator={highSchool}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('checks if the flooring function works', () => {
    const expectedFloorPercent = '42%';

    const highSchool:indicatorInfo = {
      label: 'some label',
      description: 'some description',
      type: 'percent',
      value: .426,
      isDisadvagtaged: true,
      threshold: 20,
    };
    const {asFragment} = render(
        <LocalizedComponent>
          <Indicator indicator={highSchool}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
    const container = document.querySelectorAll('li[data-cy="indicatorBox"] span');
    expect(container[0].textContent).toBe(expectedFloorPercent);
  });
});

// describe('test rendering of Indicator value icons', () => {
//   it('renders the unavailable icon when the value is null', () => {
//     const {asFragment} = render(
//         <LocalizedComponent>
//           <IndicatorValueIcon
//             value={null}
//           />
//         </LocalizedComponent>,
//     );
//     expect(asFragment()).toMatchSnapshot();
//     screen.getByAltText(EXPLORE_COPY.SIDE_PANEL_VALUES.IMG_ALT_TEXT.UNAVAILABLE.defaultMessage);
//   });
// });

describe('test rendering of Indicator value sub-text', () => {
  it('renders the "above 90 percentile"', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <IndicatorValueSubText
            value={95}
            isAboveThresh={true}
            threshold={90}
            type='percentile'
          />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders the "below 90 percentile"', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <IndicatorValueSubText
            value={89}
            isAboveThresh={false}
            threshold={90}
            type='percentile'
          />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it(`renders missing data `, () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <IndicatorValueSubText
            value={null}
            isAboveThresh={false}
            threshold={90}
            type='percentile'
          />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('test that the unit suffix renders correctly', ()=> {
  it('renders correctly when the value is a percentile', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <IndicatorValue
            type={'percentile'}
            displayStat={90}
          />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly when the value is a percent', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <IndicatorValue
            type={'percent'}
            displayStat={90}
          />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly when the value is a null', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <IndicatorValue
            type={'percentile'}
            displayStat={null}
          />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('renders value correctly for historic underinvest.', () => {
  it('checks if it renders when HRS_ET = true', () => {
    const historicUnderinvest:indicatorInfo = {
      label: 'some label',
      description: 'some description',
      type: 'boolean',
      value: true,
      isDisadvagtaged: true,
    };
    const {asFragment} = render(
        <LocalizedComponent>
          <Indicator indicator={historicUnderinvest}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('checks if it renders when HRS_ET = false:', () => {
    const historicUnderinvest:indicatorInfo = {
      label: 'some label',
      description: 'some description',
      type: 'boolean',
      value: false,
      isDisadvagtaged: true,
    };
    const {asFragment} = render(
        <LocalizedComponent>
          <Indicator indicator={historicUnderinvest}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('checks if it renders nothin when HRS_ET = null:', () => {
    const historicUnderinvest:indicatorInfo = {
      label: 'some label',
      description: 'some description',
      type: 'boolean',
      value: null,
      isDisadvagtaged: true,
    };
    const {asFragment} = render(
        <LocalizedComponent>
          <Indicator indicator={historicUnderinvest}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('renders value correctly for abandoned land mines', () => {
  it('checks if it renders when AML_RAW = true', () => {
    const abandonMines:indicatorInfo = {
      label: 'some label',
      description: 'some description',
      type: 'boolean',
      value: true,
      isDisadvagtaged: true,
    };
    const {asFragment} = render(
        <LocalizedComponent>
          <Indicator indicator={abandonMines}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('checks if it renders when AML_RAW = false:', () => {
    const abandonMines:indicatorInfo = {
      label: 'some label',
      description: 'some description',
      type: 'boolean',
      value: false,
      isDisadvagtaged: true,
    };
    const {asFragment} = render(
        <LocalizedComponent>
          <Indicator indicator={abandonMines}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('checks if it renders nothin when AML_RAW = null:', () => {
    const abandonMines:indicatorInfo = {
      label: 'some label',
      description: 'some description',
      type: 'boolean',
      value: null,
      isDisadvagtaged: true,
    };
    const {asFragment} = render(
        <LocalizedComponent>
          <Indicator indicator={abandonMines}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('renders value correctly for Former defense sites', () => {
  it('checks if it renders when FUDS_RAW = true', () => {
    const formerDefSites:indicatorInfo = {
      label: 'some label',
      description: 'some description',
      type: 'boolean',
      value: true,
      isDisadvagtaged: true,
    };
    const {asFragment} = render(
        <LocalizedComponent>
          <Indicator indicator={formerDefSites}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('checks if it renders when FUDS_RAW = false:', () => {
    const formerDefSites:indicatorInfo = {
      label: 'some label',
      description: 'some description',
      type: 'boolean',
      value: false,
      isDisadvagtaged: true,
    };
    const {asFragment} = render(
        <LocalizedComponent>
          <Indicator indicator={formerDefSites}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('checks if it renders nothin when FUDS_RAW = null:', () => {
    const formerDefSites:indicatorInfo = {
      label: 'some label',
      description: 'some description',
      type: 'boolean',
      value: null,
      isDisadvagtaged: true,
    };
    const {asFragment} = render(
        <LocalizedComponent>
          <Indicator indicator={formerDefSites}/>
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
