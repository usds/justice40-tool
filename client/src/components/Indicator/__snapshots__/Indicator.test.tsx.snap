// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of the Indicator checks if component renders 1`] = `
<DocumentFragment>
  <li
    data-cy="indicatorBox"
    data-testid="indicator-box"
  >
    <div>
      <div>
        some label
        <div>
          some description
        </div>
      </div>
      <div>
        <div>
          <div>
            <span>
              97%
            </span>
          </div>
        </div>
        <div>
          <div>
            above 
            <span>
              20%
            </span>
             percent
          </div>
        </div>
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`rendering of the Indicator checks if the flooring function works 1`] = `
<DocumentFragment>
  <li
    data-cy="indicatorBox"
    data-testid="indicator-box"
  >
    <div>
      <div>
        some label
        <div>
          some description
        </div>
      </div>
      <div>
        <div>
          <div>
            <span>
              42%
            </span>
          </div>
        </div>
        <div>
          <div>
            above 
            <span>
              20%
            </span>
             percent
          </div>
        </div>
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`renders value correctly for Former defense sites checks if it renders nothin when FUDS_RAW = null: 1`] = `
<DocumentFragment>
  <li
    data-cy="indicatorBox"
    data-testid="indicator-box"
  >
    <div>
      <div>
        some label
        <div>
          some description
        </div>
      </div>
      <div>
        <div>
          <div>
            --
          </div>
        </div>
        <div>
          <div>
            missing data
          </div>
        </div>
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`renders value correctly for Former defense sites checks if it renders when FUDS_RAW = false: 1`] = `
<DocumentFragment>
  <li
    data-cy="indicatorBox"
    data-testid="indicator-box"
  >
    <div>
      <div>
        some label
        <div>
          some description
        </div>
      </div>
      <div>
        <div>
          <div>
            No
          </div>
        </div>
        <div />
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`renders value correctly for Former defense sites checks if it renders when FUDS_RAW = true 1`] = `
<DocumentFragment>
  <li
    data-cy="indicatorBox"
    data-testid="indicator-box"
  >
    <div>
      <div>
        some label
        <div>
          some description
        </div>
      </div>
      <div>
        <div>
          <div>
            Yes
          </div>
        </div>
        <div />
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`renders value correctly for abandoned land mines checks if it renders nothin when AML_RAW = null: 1`] = `
<DocumentFragment>
  <li
    data-cy="indicatorBox"
    data-testid="indicator-box"
  >
    <div>
      <div>
        some label
        <div>
          some description
        </div>
      </div>
      <div>
        <div>
          <div>
            --
          </div>
        </div>
        <div>
          <div>
            missing data
          </div>
        </div>
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`renders value correctly for abandoned land mines checks if it renders when AML_RAW = false: 1`] = `
<DocumentFragment>
  <li
    data-cy="indicatorBox"
    data-testid="indicator-box"
  >
    <div>
      <div>
        some label
        <div>
          some description
        </div>
      </div>
      <div>
        <div>
          <div>
            No
          </div>
        </div>
        <div />
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`renders value correctly for abandoned land mines checks if it renders when AML_RAW = true 1`] = `
<DocumentFragment>
  <li
    data-cy="indicatorBox"
    data-testid="indicator-box"
  >
    <div>
      <div>
        some label
        <div>
          some description
        </div>
      </div>
      <div>
        <div>
          <div>
            Yes
          </div>
        </div>
        <div />
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`renders value correctly for historic underinvest. checks if it renders nothin when HRS_ET = null: 1`] = `
<DocumentFragment>
  <li
    data-cy="indicatorBox"
    data-testid="indicator-box"
  >
    <div>
      <div>
        some label
        <div>
          some description
        </div>
      </div>
      <div>
        <div>
          <div>
            --
          </div>
        </div>
        <div>
          <div>
            missing data
          </div>
        </div>
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`renders value correctly for historic underinvest. checks if it renders when HRS_ET = false: 1`] = `
<DocumentFragment>
  <li
    data-cy="indicatorBox"
    data-testid="indicator-box"
  >
    <div>
      <div>
        some label
        <div>
          some description
        </div>
      </div>
      <div>
        <div>
          <div>
            No
          </div>
        </div>
        <div />
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`renders value correctly for historic underinvest. checks if it renders when HRS_ET = true 1`] = `
<DocumentFragment>
  <li
    data-cy="indicatorBox"
    data-testid="indicator-box"
  >
    <div>
      <div>
        some label
        <div>
          some description
        </div>
      </div>
      <div>
        <div>
          <div>
            Yes
          </div>
        </div>
        <div />
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`test rendering of Indicator value sub-text renders missing data  1`] = `
<DocumentFragment>
  <div>
    missing data
  </div>
</DocumentFragment>
`;

exports[`test rendering of Indicator value sub-text renders the "above 90 percentile" 1`] = `
<DocumentFragment>
  <div>
    above 90
    <sup
      style="top: -0.2em;"
    >
      th
    </sup>
     percentile
  </div>
</DocumentFragment>
`;

exports[`test rendering of Indicator value sub-text renders the "below 90 percentile" 1`] = `
<DocumentFragment>
  <div>
    below 90
    <sup
      style="top: -0.2em;"
    >
      th
    </sup>
     percentile
  </div>
</DocumentFragment>
`;

exports[`test that the unit suffix renders correctly renders correctly when the value is a null 1`] = `
<DocumentFragment>
  --
</DocumentFragment>
`;

exports[`test that the unit suffix renders correctly renders correctly when the value is a percent 1`] = `
<DocumentFragment>
  <span>
    90%
  </span>
</DocumentFragment>
`;

exports[`test that the unit suffix renders correctly renders correctly when the value is a percentile 1`] = `
<DocumentFragment>
  90
  <sup
    style="top: -0.2em;"
  >
    th
  </sup>
</DocumentFragment>
`;
